src/recipe/recipe.controller.ts

```diff
@@ -75,9 +84,18 @@ export class RecipeController {
   @ApiBearerAuth()
   @ApiUnauthorizedResponse({ description: 'User is not authenticated' })
   @Get()
-  async fetchRecipes(): Promise<FetchRecipesResponse> {
-    const fetchedRecipes = await this.recipeService.fetchAllRecipes();
-
+  async fetchRecipes(
+    @UserId() userId: number,
+    @Query('authorId') authorId?: string,
+  ): Promise<FetchRecipesResponse> {
+    if (authorId) {
+      const fetchedRecipes = await this.recipeService.fetchRecipesByAuthorId(
+        +authorId,
```

Comment:
check if you can transform it to number via class-validator, so that here you can have `authorId?: number` instead of `string`